src/commands/generateFunctions.ts

```diff
@@ -0,0 +1,241 @@
+    const language = await ext.ui.showQuickPick(
+        languages.map((s) => {return { label: s, description: '', detail: '' }; }), { placeHolder: "Select language", canPickMany: false});
+
+    if (!await checkEnvironmentInstalled(language.label)) {
+        throw new Error(`You haven't installed '${language.label}' on your machine, please install '${language.label}' to continue.`);
+    }
+
+    let namespace = "";
+    if (language.label === languageTypes.Java) {
+        namespace = await askJavaNamespace();
+    } else if (language.label === languageTypes.CSharp) {
+        namespace = await askCSharpNamespace();
+    }
+
+    const uris = await askFolder();
+
+    const openAPIFilePath = path.join(uris[0].fsPath, `${node!.apiContract.name}.json`);
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("openAPI", `Downloading OpenAPI Document for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            await fse.writeFile(openAPIFilePath, openAPIDocString);
+        }
+    ).then(async () => {
+        window.showInformationMessage(localize("openAPIDownloaded", `Downloaded OpenAPI Document for API '${node!.apiContract.name} successfully.`));
+    });
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("generateFunctions", `Generating functions for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            const args: string[] = [];
+            args.push(`--input-file:${openAPIFilePath} --version:3.0.6314`);
+            args.push(`--output-folder:${uris[0].fsPath}`);
+
+            switch (language.label) {
+                case languageTypes.TypeScript:
+                    args.push('--azure-functions-typescript');
+                    args.push('--no-namespace-folders:True');
+                    break;
+                case languageTypes.CSharp:
+                    args.push(`--namespace:${namespace}`);
+                    args.push('--azure-functions-csharp');
+                    break;
+                case languageTypes.Java:
+                    args.push(`--namespace:${namespace}`);
+                    args.push('--azure-functions-java');
+                    break;
+                case languageTypes.Python:
+                    args.push('--azure-functions-python');
+                    args.push('--no-namespace-folders:True');
+                    args.push('--no-async');
+                    break;
+                default:
+                    throw new Error(localize("notSupported", "Not a supported language. We currently support C#, Java, Python, and Typescript"));
+            }
+
+            ext.outputChannel.show();
+            await cpUtils.executeCommand(ext.outputChannel, undefined, 'autorest', ...args);
```

Comment:
I liked the Functions approach of not to forcefully open the output window - Devs might be working on the terminal and might not like the forceful switch to the output window. Let's make it an option to open the output window - @miaojiang - confirming the UX here.

```typescript
const title: string = localize('generatingFunctions', 'Generating from OpenAPI...Check [output window](command:{0}) for status.', ext.prefix + '.showOutputChannel');

await window.withProgress({ location: ProgressLocation.Notification, title }, async () => {
    await cpUtils.tryExecuteCommand(ext.outputChannel, undefined, 'autorest', ...args);
});
```
![image](https://user-images.githubusercontent.com/1047040/95420856-148f2c80-08f1-11eb-8983-5340cda278a6.png)